pkg/consts/annotations.go

```diff
@@ -0,0 +1,8 @@
+package consts
+
+const (
+	// AnnotationPodTemplateSpecHash is the annotation used to store the hash of the
+	// spec of the DataPlane. This is used to detect changes in the spec of the
+	// DataPlane and trigger a rolling update.
+	AnnotationPodTemplateSpecHash = "gateway-operator.konghq.com/spec-hash"
```

Comment:
It looks like we are duplicating this annotation key with the one in `controller/dataplane/const.go`. Can we get rid of one of these?